Bug fix of era dependent configuration of MVAIso Tau-IDs

mbluj opened the pull request:
As title says: it is a fix of a bug in era dependent configuration of MVAIso Tau-IDs as reported in https://github.com/cms-nanoAOD/cmssw/issues/205.

To fix the bug a default configuration for MVAIso tau-IDs in NanoAOD is created. It is the same as used to be used exclusively for the `run2_nanoAOD_94X2016` and `run2_nanoAOD_94XMiniAODv2` eras. It is because a default configuration of taus in MiniAOD since 10_0_X is identical as for those eras in 9_4_X.

It was tested that results with the updated tau configuration in NanoAOD reproduces results of the old one for the `run2_nanoAOD_94X2016` and `run2_nanoAOD_94XMiniAODv2` eras and solves the issue for the `Run2_2018` era used for 2018 productions.

FYI, @steggema, @rmanzoni, @swozniewski

mbluj committed:
Bug fix of era dependent configuration of MVAIso Tau-ID

peruzzim commented on PhysicsTools/NanoAOD/python/taus_cff.py:
```diff
@@ -12,16 +12,12 @@
 finalTaus = cms.EDFilter("PATTauRefSelector",
     src = cms.InputTag("slimmedTausUpdated"),
-    cut = cms.string("pt > 18 && tauID('decayModeFindingNewDMs') && (tauID('byLooseCombinedIsolationDeltaBetaCorr3Hits') || tauID('byVLooseIsolationMVArun2v1DBnewDMwLT') || tauID('byVLooseIsolationMVArun2v1DBdR03oldDMwLT') || tauID('byVVLooseIsolationMVArun2v1DBoldDMwLT') || tauID('byVVLooseIsolationMVArun2v1DBoldDMwLT2017v2') || tauID('byVVLooseIsolationMVArun2v1DBnewDMwLT2017v2') || tauID('byVV ...
+    cut = cms.string("pt > 18 && tauID('decayModeFindingNewDMs') && (tauID('byLooseCombinedIsolationDeltaBetaCorr3Hits') || tauID('byVLooseIsolationMVArun2v1DBoldDMwLT2015') || tauID('byVLooseIsolationMVArun2v1DBnewDMwLT') || tauID('byVLooseIsolationMVArun2v1DBdR03oldDMwLT') || tauID('byVVLooseIsolationMVArun2v1DBoldDMwLT') || tauID('byVVLooseIsolationMVArun2v1DBoldDMwLT2017v2') || tauID('byVVLoo ...
```
This changes the tau selection in 10_2_X and above to include byVLooseIsolationMVArun2v1DBoldDMwLT2015 in the OR, is this intended?

mbluj replied:
Yes, it is intended. The changes are to make the default configuration, i.e. one for each era not explicitly specified in era-dependent customizations, identical to what was previously used exclusively for the `run2_nanoAOD_94XMiniAODv2` and `run2_nanoAOD_94X2016` era.

mbluj committed:
Merged master-102X from repository cms-nanoAOD with cms-merge-topic

mbluj committed:
add back MVAIso TauIDs to DQM

mbluj commented:
The most recent commits sync with `master-102X` branch as for today (7/11/2018) and add back MVAIso Tau-Ids to nano-DQM on top of it. This addresses comment: https://github.com/cms-nanoAOD/cmssw/pull/214#issuecomment-436283282
